[feat/allIssue#2] Get all info api

Taeg92 opened the pull request:
## 설명
> PR에 대한 간략한 설명을 작성합니다.
> (필수) 코드를 작성한 이유를 추가한다.

- 이슈 목록 상단에 필요한 open/count 개수 및 모달에 필요한 모든 데이터를 제공하는 api 추가
-  model
    - user
        - 모든 유저를 조회하는 코드 추가
    - label
        - 모든 라벨을 조회하는 코드 추가
    - milestone
        - 모든 milestone을 조회하는 코드 추가
    - issue
        - 모든 issue를 조회하는 코드 추가
        - closed된 issue를 조회하는 코드 추가
        - open된 issue를 조회하는 코드 추가

## 체크리스트
> PR 작성자와 확인하는 리뷰어가 확인해야 할 체크리스트를 작성합니다.

- [x] openCount가 정상적인 값을 반환하는 지 확인 부탁드립니다.
- [x] closedCount가 정상적인 값을 반환하는 지 확인 부탁드립니다.
- [x] data가 정상적인 값을 반환하는 지 확인 부탁드립니다.
    - [x] user 정보 확인
    - [x] label 정보 확인
    - [x] milestone 정보 확인

## 참고자료
> 해당 PR과 관련된 참고자료가 있을 경우 첨부합니다.
  
[sequelize 학습 자료](https://velog.io/@cadenzah/sequelize-document-1)

## 기타
> 리뷰어에게 상세한 설명이 필요할 경우 추가 자료를 첨부합니다. (ex. 스크린샷)

Taeg92 committed:
Feat: Implement get all info api
- 이슈 목록 상단에 필요한 open/count 개수 및 모달에 필요한 데이터를
  제공하는 api
- model label에 모든 라벨을 조회하는 코드 추가
- model issue에 closed된 issue와 open issue를 찾는 코드를 추가 이후
  count 로직에 사용
- model milestone에 모든 milestone을 조회하는 코드 추가
- model user에 모든 user를 조회하는 코드 추가
- api/all로 api 제공

Park-SooYeon commented on server/models/issue.js:
```diff
@@ -0,0 +1,57 @@
+
+    return issues;
+  } catch (err) {
+    throw new Error('유저 데이터 find 실패');
```
에러 발생 위치를 찾는 측면에서 함수의 용도에 맞게 에러 메세지를 변경해주면 더 좋을거 같아요.

sbyeol3 replied:
저희 에러메시지를 객체로 관리해서 갖다 쓰는 식으로 하면 좋을 것 같아요

Taeg92 replied:
commit e890e95
에러 메시지 객체로 관리하도록 수정 했습니다.

Park-SooYeon commented on server/services/index.js:
```diff
@@ -0,0 +1,31 @@
+
+const findAllIssues = async (req, res, next) => {
+  try {
+    const openIssue = await findIssueOpen();
+    const closedIssue = await findIssueClosed();
+    const openCount = [...openIssue].length;
+    const closedCount = [...closedIssue].length;
```
model에서 open된 이슈와 close된 이슈들의 정보를 모두 배열로 데이터를 찾아와서 배열의 length로 이슈 개수를 구한 특별한 이유가 있나요??

model에서 쿼리문의 count 함수로 open되거나 close된 이슈들의 개수를 구해서 바로 개수를 반환받을 수 있게 할 수도 있을거 같다는 생각이 드네요.

sbyeol3 replied:
여기서는 스프레드로 안 펼쳐도 되지 않을까용?

Park-SooYeon replied:
sequelize에서 count함수를 지원해줘서 issueTable.count()로 바로 개수를 반환받을 수 있더라구요,
그래서 그방법을 사용하면 배열의 length를 구하지 않고도 model에서 바로 개수를 반환 받을 수 있을거 같아서 여쭤보았습니다:)

[sequelize count](https://sequelize.org/v5/manual/models-usage.html#-code-count--code----count-the-occurrences-of-elements-in-the-database)

Taeg92 replied:
commit 5d795c1
count함수를 사용해서 코드 수정했습니다.

Park-SooYeon commented on server/app.js:
```diff
@@ -29,6 +30,7 @@ app.use(helmet());
 passportConfig();
 
 app.use(router);
+app.use('/api', apiRouter);
```
api 라우터를 추가하고 controller에서 분기해주는 방식 좋은거 같아요!

sbyeol3 replied:
저도 이렇게 했는데 컨플릭트 나겠네요 ...ㅋㅋㅋㅋㅋㅋㅋ

Park-SooYeon reviewed (commented):
API 명세서에 맞게 잘 작성해주셨네요!:)

sbyeol3 commented on server/controllers/api/index.js:
```diff
@@ -0,0 +1,8 @@
+
+const router = express.Router();
+
+router.get('/all', findAllIssues);
```
저는 아예 다 서비스로 빼지는 않았는데 이거는 논의해봐야 할 것 같습니다~

sbyeol3 reviewed (commented):
대체적으로 수연님과 의견 비슷합니다 :) 서비스 함수가 어디까지 책임을 가질지 논의해보면 좋을 것 같습니다

Taeg92 committed:
Feat: Add error messages
- error message들을 객체로 관리하기 위해 코드 작성

Taeg92 committed:
Refactor: code refactoring based on reviewers comment
- success error message 객체 추가
- open count만 반환하는 코드로 수정
- close count만 반환하는 코드로 수정

Taeg92 committed:
Refactor: Modify success message naming
- successToRead -> read로 naming 수정
